server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -196,30 +197,69 @@ SnapshotInfo finalizeSnapshot(SnapshotId snapshotId, List<IndexId> indices, long
      * <p>
      * As snapshot process progresses, implementation of this method should update {@link IndexShardSnapshotStatus} object and check
      * {@link IndexShardSnapshotStatus#isAborted()} to see if the snapshot process should be aborted.
-     * @param shard               shard to be snapshotted
+     * @param indexShard          the shard to be snapshotted
+     * @param snapshotId          snapshot id
+     * @param indexId             id for the index being snapshotted
+     * @param snapshotIndexCommit commit point
+     * @param snapshotStatus      snapshot status
+     * @deprecated use {@link #snapshotShard(Store, MapperService, SnapshotId, IndexId, IndexCommit, IndexShardSnapshotStatus)} instead
```

Comment:
I wonder if we even need this deprecation? We will need to make changes to this interface anyway to fix https://github.com/elastic/elasticsearch/issues/41581 so maybe it's not worth being careful here when we have to break 3rd party implementations (not sure if there even are any) to fix functionality anyway?

Reply:
Why not drop the remaining deprecations as well 4a1bd35ffa2 ? :)

Reply:
well I can drop it in a followup but I want to backport this change to 7.x and there the deprecation is mandatory IMO

Reply:
Interesting :) I don't wanna take this one off-topic but I have to ask:

If we don't want to break the `Repository` interface in `7.x` that makes fixing https://github.com/elastic/elasticsearch/issues/41581 in `7.x` effectively impossible. We simply need to change what initialize snapshot and finalize snapshot do to get that one right.
So if we really can't break the contract for that one, we'd have to add new versions of initialize and finalize and keep the deprecated ones with broken behavior around?
(note that we also made a number of breaking changes to `BlobStoreRepository` itself lately and are planning to make even bigger ones in https://github.com/elastic/elasticsearch/pull/42189. Those would break any 3rd party implementation of `BlobStoreRepository` as well. Should we tread more carefully there than planned?

Reply:
I personally see this being a different reason. If you need to break an interface for the sake of fixing a bug or a correctness issue we should and can break it in a minor release. If we just factor stuff out to make it more testable I think being nice to a user is a wise decision. I will for sure remove the deprecated methods from master once it's backported.